libsignal-service/src/push_service.rs

```diff
@@ -292,15 +292,11 @@ pub enum ServiceError {
 }
 
 #[async_trait::async_trait(?Send)]
-pub trait PushService {
+pub trait PushService: Clone {
```

Comment:
I'm not sure whether I like to really *enforce* `: Clone`.  This is only used in the provisioning manager.  What about passing the provisioning manager the `service` by (mutable) reference? Can't we generically `impl PushService for &mut PushService` or something like that?

Reply:
If you still want to go the `: Clone` route, maybe just put the trait bound on the provisioningmanager impl block, or even on that single method.

Reply:
Let me try the `&mut` approach.

Reply:
It worked. Please see the rebased review.